proto/ibc/core/channel/v1/tx.proto

```diff
@@ -279,18 +279,14 @@ message MsgChannelUpgradeTry {
   option (gogoproto.equal)           = false;
   option (gogoproto.goproto_getters) = false;
 
-  string                    port_id                  = 1;
-  string                    channel_id               = 2;
-  Channel                   counterparty_channel     = 3 [(gogoproto.nullable) = false];
-  uint64                    counterparty_sequence    = 4;
-  Channel                   proposed_upgrade_channel = 5 [(gogoproto.nullable) = false];
-  ibc.core.client.v1.Height timeout_height           = 6 [(gogoproto.nullable) = false];
-  uint64                    timeout_timestamp        = 7;
-  bytes                     proof_channel            = 8;
-  bytes                     proof_upgrade_timeout    = 9;
-  bytes                     proof_upgrade_sequence   = 10;
-  ibc.core.client.v1.Height proof_height             = 11 [(gogoproto.nullable) = false];
-  string                    signer                   = 12;
+  string                    port_id                       = 1;
+  string                    channel_id                    = 2;
+  Upgrade                   proposed_upgrade              = 3 [(gogoproto.nullable) = false];
```

Comment:
why actually do we need both proposed upgrades? why not just the counterparty?

Reply:
The try step is verifying the counterparty proposed upgrade and setting its own values for the upgrade. Note that the modifiable fields must be the same but that the other fields should be unique to the side upgrading (last packet sent, timeout)